src/mod/self.c

```diff
@@ -442,13 +437,11 @@ _cleanup_thread(pthread_t pid)
 
         struct self *self = container_of(item, struct self, retired_node);
 
-        if ((uint64_t)self->pid != (uint64_t)pid)
+        if ((pid != 0 && ((uint64_t)self->pid != (uint64_t)pid)) ||
+            (pid == 0 && !_thread_dead(self)))
```

Comment:
This condition is a bit confusing. Can we flip it? If `(pid != 0 && ((uint64_t)self->pid == (uint64_t)pid)) || (pid == 0 && _thread_dead(self) )`

Reply:
Ok